src/app/reporting/Engine.cpp

```diff
@@ -100,15 +108,38 @@ CHIP_ERROR Engine::BuildSingleReportDataAttributeDataList(ReportData::Builder &
     {
         if (clusterInfo->IsDirty())
         {
-            AttributeDataElement::Builder attributeDataElementBuilder = attributeDataList.CreateAttributeDataElementBuilder();
             if (apReadHandler->IsInitialReport())
             {
                 // Retrieve data for this cluster instance and clear its dirty flag.
-                err = RetrieveClusterData(attributeDataElementBuilder, *clusterInfo);
+                err = RetrieveClusterData(attributeDataList, *clusterInfo);
                 VerifyOrExit(err == CHIP_NO_ERROR,
                              ChipLogError(DataManagement, "<RE:Run> Error retrieving data from cluster, aborting"));
                 attributeClean = false;
             }
+            else
+            {
+                for (auto & path : mDirtyPathList)
+                {
+                    if (path.IsDirty())
+                    {
+                        if (clusterInfo->IsAttributePathSupersetOf(path))
+                        {
+                            err = RetrieveClusterData(attributeDataList, path);
+                        }
+                        else if (path.IsAttributePathSupersetOf(*clusterInfo))
+                        {
+                            err = RetrieveClusterData(attributeDataList, *clusterInfo);
+                        }
+                        else
+                        {
+                            continue;
```

Comment:
is superset the only option? It may be worth mentioning in a comment: partial overlap is not possible, hence the 'continue' here: clusterInfo and path have nothing in common.

Reply:
done, thanks